### PR TITLE
Adding centos 6.9 based image

### PR DESCRIPTION
**Pull request by astrilet**
adding centos 6.9 image

### CODE REVIEW

File: rubyplus-centos6/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+FROM centos:centos6.9
```
Comment (murny) [inline rubyplus-centos6/Dockerfile]: Yes exactly. Its based off another docker image. This one is the official centos image: https://hub.docker.com/_/centos/

File: rubyplus-centos6/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+ADD files/RPM-GPG-KEY-CentOS-Debug-6  /etc/pki/rpm-gpg
+ADD files/RPM-GPG-KEY-CentOS-Security-6 /etc/pki/rpm-gpg
+ADD files/RPM-GPG-KEY-CentOS-Testing-6  /etc/pki/rpm-gpg
+ADD files/RPM-GPG-KEY-EPEL-6  /etc/pki/rpm-gpg
+
+#run update
+RUN yum -y update
+
+#install required packages
+RUN yum install -y curl wget bzip2 redis resque nodejs libreoffice ImageMagick-devel \
```
Comment (murny) [inline rubyplus-centos6/Dockerfile]: Not all ruby projects at ualib will need all this extra things. Not all ruby projects will need resque/codecs/fits/phantomjs. This should be application specific and live in the application image. Same with the rubyplus image...its currently installing java (oracle java too, not openjdk). Why does a ruby image need to come with java and other bloat? 

If we create a PushmiPullyu docker image and based it off one of these....PushmiPullyu simply doesn't need all these things. As it doesn't use FITS, PhantomJS, Java, etc.

File: rubyplus-centos6/files/RPM-GPG-KEY-CentOS-6 [hunk 3]
```diff
@@ -0,0 +1,30 @@
+-----BEGIN PGP PUBLIC KEY BLOCK-----
```
Comment (mbarnett) [inline rubyplus-centos6/files/RPM-GPG-KEY-CentOS-6]: This is safe. The public RPM-GPG-KEY just lets the image confirm that any RPMs it pulls are properly signed -- it can't be used to authorize or log in to anything. It's the private key that should never be published.

File: rubyplus-deb/Dockerfile [hunk 4]
```diff
@@ -2,6 +2,8 @@ FROM ualibraries/ruby2.3.3:latest
```
Comment (murny) [inline rubyplus-deb/Dockerfile]: This has been a common push back of mine on all your PRs. I still see no reason for creating a custom ruby image. What does this provide over the official ruby image?

First your basing this off ualibraries/ruby2.3.3:latest....Where is the source code for this? It isn't found in this repo. This is a big maintenance red flag. How can someone upgrade this if you can't even find the source code. How does the rest of the team get access to this dockerhub and be able to update this accordingly? We need documentation for this. Similar comment made here: https://github.com/ualbertalib/didocker_hydranorth2/pull/2#discussion_r109217455

Second the rest of the configuration in this Dockerfile is all application specific. Your installing java, fits ,phantomjs and other things. Not all ruby apps need this. Same comment on custom Centos image here:
https://github.com/ualbertalib/di_docker_images/pull/8/files#r114829435

Third point this rubyplus image is almost 800mb itself? Why is this so big? For example the official docker ruby image is only 100-300MB? what does rubyplus image give over the default ruby image? I still see no benefit of this. 

It just means we have to maintain this ourselves instead of utilizing the communities hardwork.

File: rubyplus-centos6/Dockerfile (hunk 2 above)
Comment (weiweishi) [inline rubyplus-centos6/Dockerfile]: Agreed. DMP etc won't need the extra components too, like resque, nodejs, ImageMagick etc.

File: solr/files/start_solr.sh [hunk 5]
```diff
@@ -1,17 +1,18 @@
 #!/bin/bash
```
Comment (murny) [inline solr/files/start_solr.sh]: Should be a separate PR. I am assuming this is a bug fix for something I questioned about on the hyku PR: https://github.com/ualbertalib/didocker_hyku/pull/11/files#r114023962

Regarding the Solr folder in this repo...Why do we have a HN2 submodule in this folder? Why are we pushing HN2 code into the Solr container then using it for other projects that are not HN2 (like Hyku).

File: rubyplus-deb/Dockerfile [hunk 6]
```diff
@@ -2,6 +2,8 @@ FROM ualibraries/ruby2.3.3:latest
 
 RUN echo "deb http://ppa.launchpad.net/webupd8team/java/ubuntu xenial main" | tee /etc/apt/sources.list.d/webupd8team-java.list
 RUN echo "deb-src http://ppa.launchpad.net/webupd8team/java/ubuntu xenial main" | tee -a /etc/apt/sources.list.d/webupd8team-java.list
+RUN echo "deb http://httpredir.debian.org/debian jessie-backports main non-free" | tee -a /etc/apt/sources.list
+RUN echo "deb-src http://httpredir.debian.org/debian jessie-backports main non-free" | tee -a /etc/apt/sources.list
 RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886
 
 RUN echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections
```
Comment (mbarnett) [inline rubyplus-deb/Dockerfile]: I think we should use the OpenJDK instead of Oracle. Especially if we distribute this to others, auto-accepting the Oracle license on their behalf is problematic. We also don't really gain anything by using Oracle over the open-source version.

Java also doesn't belong in the base image as we don't need it for all of our projects.

File: rubyplus-deb/Dockerfile [hunk 7]
```diff
@@ -29,4 +31,15 @@ RUN mkdir -p /usr/local/fits \
     && chmod a+x /usr/local/fits/fits-1.0.6/fits.sh \
     && ln -s /usr/local/fits/fits-1.0.6/fits.sh /usr/bin/fits
 
+RUN cd / \
+    && export PHANTOM_JS="phantomjs-1.9.8-linux-x86_64" \
```
Comment (mbarnett) [inline rubyplus-deb/Dockerfile]: I don't think phantomjs belongs in a base image, as we don't use it in all of our projects (PushmiPullyu doesn't need it, for example).

File: rubyplus-deb/Dockerfile [hunk 8]
```diff
@@ -29,4 +31,15 @@ RUN mkdir -p /usr/local/fits \
     && chmod a+x /usr/local/fits/fits-1.0.6/fits.sh \
```
Comment (mbarnett) [inline rubyplus-deb/Dockerfile]: Fits doesn't belong in the base image, as we don't need it in all of our projects

File: rubyplus-deb/Dockerfile [hunk 9]
```diff
@@ -16,10 +18,10 @@ RUN set -ex \
 		nano-tiny \
```
Comment (mbarnett) [inline rubyplus-deb/Dockerfile]: Do we need a text editor in this image?

File: rubyplus-centos6/Dockerfile (hunk 2 above)
Comment (astrilet) [inline rubyplus-centos6/Dockerfile]: Good point will move them to the app specific image

File: rubyplus-deb/Dockerfile (hunk 4 above)
Comment (astrilet) [inline rubyplus-deb/Dockerfile]: 1. Idea was to make environment as close as possible to what we have in production. 
2. I agree that my base image contains some software that should not be there, will move them to the app specific docker file.
3. All additional software that is installed make image bigger (top 20 packages by size)
 271062057 - jdk1.8.0_102              	 1.8.0_102
 269929713 - libreoffice-core          	 4.3.7.2
 112436037 - glibc-common              	 2.12
  82633502 - ualib-rails               	 4.2.7
  65581081 - ffmpeg                    	 0.1
  64996605 - libgcj                    	 4.4.7
  37912260 - ualib_ruby                	 2.3.3
  35282834 - perl                      	 5.10.1
  29517275 - libreoffice-calc          	 4.3.7.2
  22979382 - python-libs               	 2.6.6
  22395193 - mesa-private-llvm         	 3.6.2
  20971477 - ghostscript               	 8.70
  19519528 - libicu                    	 4.2.1
  17728852 - libreoffice-writer        	 4.3.7.2
  15816517 - redhat-logos              	 60.0.14
  14110408 - mesa-dri1-drivers         	 7.11
  13381176 - mesa-dri-drivers          	 11.0.7
  13121335 - glibc                     	 2.12
  12873032 - coreutils                 	 8.4
  12426003 - gtk2                      	 2.24.23

Comment (murny) [inline rubyplus-deb/Dockerfile]: So by doing step 2 (removing the software that doesn't need to be here and instead live in the app image) you basically solve step 3 (or definitely help cut down this image size).

But step 1 is still a question and hasn't been answered. Where is source code for `ualibraries/ruby2.3.3:latest`? Where is the documentation for dockerhub? How can anyone but yourself able to maintain this? There is zero documentation in any of the dockerhub descriptions: https://hub.docker.com/r/ualibraries

Comment (astrilet) [inline rubyplus-deb/Dockerfile]: it is basically official ruby image, will fix that (replace with official ruby image)

File: rubyplus-deb/Dockerfile (hunk 9 above)
Comment (astrilet) [inline rubyplus-deb/Dockerfile]: removed that

File: rubyplus-deb/Dockerfile (hunk 8 above)
Comment (astrilet) [inline rubyplus-deb/Dockerfile]: rename that base image to be rubyhydra (specific to hydra projects) W.e do not need our own ruby image for other projects, can use official one.

File: rubyplus-deb/Dockerfile (hunk 6 above)
Comment (astrilet) [inline rubyplus-deb/Dockerfile]: You are right, will replace that with OpenJDK

File: rubyplus-deb/Dockerfile (hunk 7 above)
Comment (astrilet) [inline rubyplus-deb/Dockerfile]: me that out of base image

File: rubyplus-centos6/Dockerfile [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+RUN cd / \
+    && export PHANTOM_JS="phantomjs-1.9.8-linux-x86_64" \
+    && wget https://bitbucket.org/ariya/phantomjs/downloads/$PHANTOM_JS.tar.bz2 \
+    && tar xvjf $PHANTOM_JS.tar.bz2 \
+    && mv $PHANTOM_JS /usr/local/share \
+    && rm $PHANTOM_JS.tar.bz2  \
+    && ln -sf /usr/local/share/$PHANTOM_JS/bin/phantomjs /usr/local/bin
+
+#install ruby and rails
+RUN yum install -y ualib_ruby ualib-rails
```
Comment (astrilet) [inline rubyplus-centos6/Dockerfile]: for centos 6 we will use ruby from our private rpm repositories, for debinan base image it will be based on community released ruby

File: rubyplus-centos6/Dockerfile [hunk 11]
```diff
@@ -0,0 +1,46 @@
+FROM centos:centos6.9
+MAINTAINER Alex Strilets <strilets@ualberta.ca>
+
+# add out custom repos
```
Comment (astrilet) [inline rubyplus-centos6/Dockerfile]: for centos 6 we will use our private rpm repositories to be as close to prod environment as possible, for debian base image it will be based on community released ruby

File: rubyplus-centos6/Dockerfile (hunk 1 above)
Comment (astrilet) [inline rubyplus-centos6/Dockerfile]: yes it is based on official centos 6.9 released image

File: solr/files/start_solr.sh (hunk 5 above)
Comment (astrilet) [inline solr/files/start_solr.sh]: I will create separate PR for this change

File: rubyhydra-deb/Dockerfile [hunk 12]
```diff
@@ -15,9 +16,8 @@ RUN set -ex \
 	&& buildPkgs=' \
 		unzip \
 		vim \
-		nano-tiny \
 		git \
-    oracle-java8-installer \
+    openjdk-7-jre \
```
Comment (murny) [inline rubyhydra-deb/Dockerfile]: Now that you switched to openjdk-7, is adding the source list stuff (lines 3-8?) for oracle java and accepting the oracle license still needed then?

**Comment by weiweishi** [comment]
@astrilet This PR requires your attention, to remove https://github.com/ualbertalib/di_docker_images/blob/e9202f2a5e0a146f7a0ca8b55c99dfe449b3dd33/rubyhydra-deb/Dockerfile#L10-L11

**Comment by astrilet** [comment]
This was a typo, this image will only be used to create HydraNorth

**Comment by weiweishi** [comment]
@astrilet When you have a moment, please add some documentation to dockerhub for these two docker images too, as those are part of the work this PR addresses.

File: rubyhydra-deb/README.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
+# rubyhydra-dev image
```
Comment (weiweishi) [inline rubyhydra-deb/README.md]: Could you also add links to these two images on dockerhub here? And on the documentation in dockerhub, add links back to the dockerfiles.

File: rubyplus-centos6/README.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# rubyplus-centos image
```
Comment (weiweishi) [inline rubyplus-centos6/README.md]: Same here re: link to the docker images.

**Comment by weiweishi** [comment]
@astrilet I suggest you to take a look at the README template @jefferya created:
https://github.com/ualbertalib/pushmi_pullyu/blob/issue_25_docker_image/docker/README.md
And fill in the sections that you see applicable in these docker images.

File: rubyhydra-deb/Dockerfile [hunk 15]
```diff
@@ -1,25 +1,24 @@
-FROM ualibraries/ruby2.3.3:latest
+FROM ruby:2.3.4
 
 RUN echo "deb http://ppa.launchpad.net/webupd8team/java/ubuntu xenial main" | tee /etc/apt/sources.list.d/webupd8team-java.list
```
Comment (murny) [inline rubyhydra-deb/Dockerfile]: Are these needed if you not using oracle java?

File: rubyhydra-deb/README.md [hunk 16]
```diff
@@ -0,0 +1,61 @@
...
+     git clone https://github.com/di_docker_images
+     ```
+  2. Go to the directory where you cloned the project
+     ```shell
+     docker build . -t ualibraries/rubyhydra
+     ```
+
+     Alternatively you can always pull pre-build image from docker hub
+     ```shell
+     dockre pull ualibraries/rubyhydra
```
Comment (murny) [inline rubyhydra-deb/README.md]: Spelling mistakes. Should be docker?

File: rubyhydra-deb/README.md [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+
+2. push to the Docker Hub registry - `docker push <hub-user>/<repo-name>:<tag>`
+    * `docker push ualibraries/rubyhydra`
+
+## Upgrading local container ###
+
+To upgrade to a newer release of rubyhydra
+
+* stop currently running image:
+  * `docker stop ${container_id}`
```
Comment (murny) [inline rubyhydra-deb/README.md]: I think alot of this was suppose to be fill in the blanks? What is `${container_id}`? It's the name of your container.

Fill in these blanks?

File: rubyplus-centos6/README.md [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+     git clone https://github.com/di_docker_images
+     ```
+  2. Go to the directory where you cloned the project
+     ```shell
+     docker build . -t ualibraries/ruby_base:centos69
+     ```
+
+     Alternatively you can always pull pre-build image from docker hub
+     ```shell
+     dockre pull ualibraries/ruby_base:centos69
```
Comment (murny) [inline rubyplus-centos6/README.md]: Here too

File: rubyhydra-deb/Dockerfile [hunk 19]
```diff
@@ -29,4 +28,8 @@ RUN mkdir -p /usr/local/fits \
     && chmod a+x /usr/local/fits/fits-1.0.6/fits.sh \
     && ln -s /usr/local/fits/fits-1.0.6/fits.sh /usr/bin/fits
 
+
+	RUN apt-get install -q -y zip \
```
Comment (murny) [inline rubyhydra-deb/Dockerfile]: Why seperate the zip package from the rest? Why not add this to the other apt-get installs on Line 11?

File: rubyhydra-deb/Dockerfile [hunk 20]
```diff
@@ -1,25 +1,24 @@
...
 RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886
-
-RUN echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections
-RUN echo debconf shared/accepted-oracle-license-v1-1 seen true | debconf-set-selections
+RUN echo 'deb http://ftp.uk.debian.org/debian jessie-backports main' >> /etc/apt/sources.list
 
 # install some usefull packages on top of bare ruby
 RUN set -ex \
   && export DEBIAN_FRONTEND=noninteractive \
 	&& buildPkgs=' \
```
Comment (murny) [inline rubyhydra-deb/Dockerfile]: is the variable needed? Why not just install them without assigning to a variable first? Especially when you are never using the variable more then once.

File: rubyhydra-deb/Dockerfile (hunk 15 above)
Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: Removed

File: rubyhydra-deb/README.md (hunk 16 above)
Comment (astrilet) [inline rubyhydra-deb/README.md]: fixed

File: rubyhydra-deb/README.md (hunk 17 above)
Comment (astrilet) [inline rubyhydra-deb/README.md]: docker assigns random names to the running containers, docker ps  will show them.

File: rubyplus-centos6/README.md (hunk 18 above)
Comment (astrilet) [inline rubyplus-centos6/README.md]: fixed

File: rubyhydra-deb/Dockerfile (hunk 19 above)
Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: moved into one place

File: rubyhydra-deb/Dockerfile (hunk 20 above)
Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: Nope, left over from oracle installation, removed.

File: rubyhydra-deb/README.md (hunk 17 above)
Comment (murny) [inline rubyhydra-deb/README.md]: This is true if you just run docker build. But we are giving them a custom name above by passing in the `-t` tag flag and calling it `ualibraries/rubyhydra`. So shouldn't all these be `ualibraries/rubyhydra`?

If not, add this to the documentation indicating how to get this `container_id` by running `docker ps`?

File: rubyhydra-deb/Dockerfile (hunk 20 above)
Comment (murny) [inline rubyhydra-deb/Dockerfile]: I was actually refering to the `buildPkgs` variable not the `DEBIAN_FRONTEND`

File: rubyhydra-deb/Dockerfile [hunk 21]
```diff
@@ -1,22 +1,20 @@
...
-RUN echo debconf shared/accepted-oracle-license-v1-1 select true | debconf-set-selections
-RUN echo debconf shared/accepted-oracle-license-v1-1 seen true | debconf-set-selections
+RUN echo 'deb http://ftp.uk.debian.org/debian jessie-backports main' >> /etc/apt/sources.list
 
 # install some usefull packages on top of bare ruby
 RUN set -ex \
-  && export DEBIAN_FRONTEND=noninteractive \
 	&& buildPkgs=' \
 		unzip \
+    zip \
```
Comment (murny) [inline rubyhydra-deb/Dockerfile]: weird alignment issues?

File: rubyhydra-deb/Dockerfile [hunk 22]
```diff
@@ -1,22 +1,20 @@
...
 		vim \
-		nano-tiny \
 		git \
-    oracle-java8-installer \
-	' \
+    openjdk-7-jre \
+    build-essential chrpath libssl-dev libxft-dev\
```
Comment (murny) [inline rubyhydra-deb/Dockerfile]: why have most of these be one per line, but this one is different?

File: rubyhydra-deb/README.md (hunk 17 above)
Comment (weiweishi) [inline rubyhydra-deb/README.md]: I think it probably would make sense to give the container a name with --name flag in the command in line 17? I found it's easier for me to keep track, restart, and log into the container with a name, without have to do ```docker ps``` every time? Then here we can just do ```docker stop rubyhydra```.

Comment (murny) [inline rubyhydra-deb/README.md]: Yeah, Agreed.

(Not sure if theres a --name flag, but the --tag/-t flag will do exactly what your looking for though https://docs.docker.com/engine/reference/commandline/build/#options)

File: rubyhydra-deb/Dockerfile (hunk 22 above)
Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: can no be perfect, do not really know this is how I wrote it, can break one per line

File: rubyhydra-deb/Dockerfile (hunk 21 above)
Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: set tabs to be 4 char

Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: switched to spaces

File: rubyhydra-deb/README.md (hunk 17 above)
Comment (weiweishi) [inline rubyhydra-deb/README.md]: Sorry I meant for the --name flag in ```docker run```. Maybe add that to line 53? Having that in the instruction so people can easily refer to the container they've created?

Comment (astrilet) [inline rubyhydra-deb/README.md]: if you name your container with --name flag that will work, but otherwise it will be random, like in the example below:
CONTAINER ID        IMAGE                   COMMAND                  CREATED             STATUS              PORTS                    NAMES
0670e9cbdfc5        ualibraries/swift_dev   "/bin/sh -c /usr/l..."   28 hours ago        Up 28 hours         0.0.0.0:8080->8080/tcp   **# dreamy_noether**

Comment (astrilet) [inline rubyhydra-deb/README.md]: naming running containers has it drawback, if you stop it you would have do delete image of it otherwise you will not be able to run another instance with the same name (pain in scripts)

Comment (weiweishi) [inline rubyhydra-deb/README.md]: From my limited use of docker container, I had to log in and out of the container, and a name allows me to run named containers again later with ```docker start```. But my use was limited so I'm fine either way.

Comment (murny) [inline rubyhydra-deb/README.md]: I know your frustrated alex but this is why good documentation is needed. So people with limited experience (or no experience) with docker can figure this out without asking questions or googling. Yes my bad (was thinking of the image and not the container). Your right, you can use `--name` when using the `docker run` command to name the container then use that name later on to do things to it (stop/restart/etc to it).

From stackoverflow (http://stackoverflow.com/questions/32073971/stopping-docker-containers-by-image-name-ubuntu):
```
You could start the container setting a container name:

docker run -d --name <container-name> <image-name>

Same image could be used to spin up multiple containers, so this is a good way to start a container. Then you could use this container-name to stop, attach... the container:

docker exec -it <container-name> bash
docker stop <container-name>
docker rm <container-name>
```

If you do not want a name on your container you can use some bash magic for looking up the image name:
```
docker stop $(docker ps -q --filter ancestor=<image-name> )
```
Regarding the `pain in scripts` issue? Can you give examples of this? Maybe there's a better way to do this then?

Either way, this kind of documentation is needed, instead of just saying to the user figure it out yourself ... use something like `docker run ...` or use `docker stop {$container_id}` and not explain any more then that

**Comment by weiweishi** [comment]
@astrilet 
two outstanding questions on this PR:

- [ ] Can we not using the buildPkgs variable, and just do it in one command:

``` 
   RUN apt-get update -qq \
    && apt-get install -y build-essential \
                          other packages \
                          more packages \
    && rm -rf /var/lib/apt/lists/*
```

It will make the code more readable. 

- [ ] what are these following lines for:
```
RUN echo "deb http://httpredir.debian.org/debian jessie-backports main non-free" | tee -a /etc/apt/sources.list
RUN echo "deb-src http://httpredir.debian.org/debian jessie-backports main non-free" | tee -a /etc/apt/sources.list
RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886
RUN echo 'deb http://ftp.uk.debian.org/debian jessie-backports main' >> /etc/apt/sources.list
```
if they are needed, could you please add documentation on what they are used for?

**Comment by astrilet** [comment]
I will remove variable and put everything in the command line;
these lines are to get ffmpeg package installed.

**Comment by weiweishi** [comment]
@astrilet could you please add an inline comment for the few lines needed for ffmpeg package installation in the Dockerfile, so other people understand what is needed for?
I will wait for your updated commits to approve the PR.

File: rubyhydra-deb/Dockerfile [hunk 23]
```diff
@@ -1,5 +1,6 @@
 FROM ruby:2.3.4
 
+# following 4 lines required to install ffmpeg package :wq
```
Comment (mbarnett) [inline rubyhydra-deb/Dockerfile]: Vim command crept in there

Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: sorry...

File: rubyhydra-deb/Dockerfile [hunk 24]
```diff
@@ -0,0 +1,32 @@
...
+RUN apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys EEA14886
+RUN echo 'deb http://ftp.uk.debian.org/debian jessie-backports main' >> /etc/apt/sources.list
+
+# install some usefull packages on top of bare ruby
+RUN set -ex \
+  && apt-get update \
+  && apt-get install -q -y unzip \
+                           zip \
+                           vim \
+                           git \
```
Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: later we need git to pull code from github in some cases

Comment (astrilet) [inline rubyhydra-deb/Dockerfile]: But if you feel we do not need it I will remove it, we can always add it in place where we need to clone the repository, thought it would be nice to have feature...
